rclcpp/include/rclcpp/create_publisher.hpp

```diff
@@ -29,6 +29,46 @@
 namespace rclcpp
 {
 
+/// Create and return a publisher of the given MessageT type.
+/**
+ * The NodeT type only needs to have a method called get_node_topics_interface()
+ * which returns a shared_ptr to a NodeTopicsInterface.
+ */
+template<
+  typename MessageT,
+  typename AllocatorT = std::allocator<void>,
+  typename PublisherT = rclcpp::Publisher<MessageT, AllocatorT>,
+  typename NodeT>
+std::shared_ptr<PublisherT>
+create_generic_publisher(
+  NodeT & node,
+  const std::string & topic_name,
+  const rosidl_message_type_support_t & type_support,
```

Comment:
Why both `MessageT` template parameter and the `type_support` argument are needed?

Reply:
Naming should also be more clear.
Maybe, just an override of `create_publisher`?

Reply:
I think MessageT is used for the allocator